pkg/api/dashboard_snapshot.go

```diff
@@ -77,7 +77,7 @@ func (hs *HTTPServer) CreateDashboardSnapshot(c *contextmodel.ReqContext) {
 
 	// Do not check permissions when the instance snapshot public mode is enabled
 	if !hs.Cfg.SnapshotPublicMode {
-		evaluator := ac.EvalPermission(dashboards.ActionDashboardsWrite, dashboards.ScopeDashboardsProvider.GetResourceScopeUID(cmd.Dashboard.GetNestedString("uid")))
+		evaluator := ac.EvalAll(ac.EvalPermission(dashboards.ActionSnapshotsCreate), ac.EvalPermission(dashboards.ActionDashboardsRead, dashboards.ScopeDashboardsProvider.GetResourceScopeUID(cmd.Dashboard.GetNestedString("uid"))))
```

Comment:
Without specific RBAC role for Snapshots we decided only users with dashboard write permissions can create a Snapshot but now, I think it's enough with `dashboards.ActionDashboardsRead` and the new `dashboards.ActionSnapshotsCreate`. 
Without RBAC this is still valid for users with edit role.

Reply:
Without RBAC this would mean that no Viewer users can ever create a snapshot, even on dashboards that they can edit, as they wouldn't have the `snapshots:create` permission.

Reply:
We decided this is ok, in OSS only editors can create snapshots.

Reply:
I removed the Scope from the `ActionSnapshotsCreate` because this action will be scoped to `ActionDashboardsRead` using dashboard uid. WDYT?